odoo/tools/safe_eval.py

```diff
@@ -380,7 +382,7 @@ def safe_eval(expr, globals_dict=None, locals_dict=None, mode="eval", nocopy=Fal
     if globals_dict is None:
         globals_dict = {}
 
-    globals_dict['__builtins__'] = _BUILTINS
+    globals_dict['__builtins__'] = _BUILTINS if globals_builtins else None
```

Comment:
Regarding this change: 
- it doesn't bring an additional security, as what is available in `_BUILTINS` is considered as safe
- Initially, this was added so to have the same builtins in both the JS evaluator and the Python evaluator. To have a "synchronization" regarding the builtins. But, it was never the case:
  - the previous implementation with the `restricted_eval` had all JS builtins (see my previous example with alert)
  - the new implementation using  `evaluateExpr` doesn't restrict the builtins to just `min` and `max`.
 
So, I would remove this change, as in my opinion it has no reason to be currently.

Reply:
I'm ok with that. Since the `_check_formula` explicitely checks the tokens used inside the formula, it already prevent accessing stuff that are not js-side like range/tuple etc